### DISCUSSION TITLE
How can I change the WhatsApp client connection name?

### DISCUSSION THREAD

**Discussion by ibrahimypr** [tulir/whatsmeow#469] [WhatsApp protocol Q&A] [answered]
How can I change the WhatsApp client connection name? By default it is Whatsmeow. I want to change it to Google Chrome.

**Comment by AltNico** [comment]
Searching for `\"whatsmeow` in this project led me to this line:

https://github.com/tulir/whatsmeow/blob/00abc29ba510ec59e329008b015dd4af061949a3/store/clientpayload.go#L126

Would be cool if this could be configurable. @tulir Would you be open to MRs for this?

**Reply by tulir** [reply]
It's a public variable, you can just change the fields inside it (or use the SetOSInfo method that's right next to it, although that doesn't have a parameter for setting the browser type)

**Reply by AltNico** [reply, marked as answer]
Yeah, I also just realized this. Sorry for the noise, @tulir!

For the records: this is the code that changes the device name:
```go
store.DeviceProps.Os = proto.String("Oculta")
```

Also see

https://github.com/tulir/whatsmeow/blob/00abc29ba510ec59e329008b015dd4af061949a3/mdtest/main.go#L60

**Reply by Honeyb5150** [reply]
> Searching for `\"whatsmeow` in this project led me to this line:
> 
> 
> 
> https://github.com/tulir/whatsmeow/blob/00abc29ba510ec59e329008b015dd4af061949a3/store/clientpayload.go#L126
> 
> 
> 
> Would be cool if this could be configurable. @tulir Would you be open to MRs for this?

@AltNico

**Comment by ibrahimypr** [comment]
thank you very much everyone

**Comment by macedo** [comment]
the connection name is the same we see in linked devices page of whatsapp? when I connect I see "OtherDevice".

**Reply by macedo** [reply]
@tulir can you help me? I need to change the name showed on linked devices session.

**Reply by till-kaemmerer** [reply]
@macedo maybe I'm too late but for me the issue was the same. Setting `store.DeviceProps.PlatformType = waProto.DeviceProps_DESKTOP.Enum()` seems to do the trick. Whatsapp seems to ignore the os name if the platform type is unknown.

**Comment by muhammadrivaldy** [comment]
Hi @ibrahimypr 

I just go deep dive into the problem, and I saw we can modified it as real connection with this piece of codes

```
store.SetOSInfo("Linux", store.GetWAVersion())
store.DeviceProps.PlatformType = waCompanionReg.DeviceProps_CHROME.Enum()
```

**Reply by till-kaemmerer** [reply]
@ibrahimypr I found that `DESKTOP` works even better because it then just shows the os name in Whatsapp and not "Google Chrome (os name)"

**Reply by AlejoGarat** [reply]
Hi @till-kaemmerer!
Your comment suggests using DESKTOP for a cleaner display. My primary goal is to prevent frequent disconnections and sudden logouts. Have you experimented with setting PlatformType to ANDROID or IOS (to match the main device OS) instead of DESKTOP? If so, did you find that the ANDROID/IOS setting resulted in more disconnections or forced session closures compared to using DESKTOP? We are trying to determine if the community's preference for DESKTOP is purely for display/aesthetics or if it genuinely offers superior connection stability by preventing WhatsApp from confusing the companion client with a second mobile phone.

**Reply by till-kaemmerer** [reply]
@AlejoGarat In my case it was solely for the sake of a cleaner display for our users. I don't actually know if the PlatformType has any impact on connectivity (maybe @tulir can give an answer to that). I have not tested connectivity with other platform types but it would be interesting to know if it leads to less disconnections and/or closures.
If you have any findings I would much appreciate the insight.

**Reply by AlejoGarat** [reply]
Thank you @till-kaemmerer! Do you have any recommendation about it @tulir?
